frontend/src/components/assessments/student-experience/Question.tsx

```diff
@@ -9,62 +9,76 @@ import {
   QuestionTextMetadata,
   TextMetadata,
 } from "../../../types/QuestionMetadataTypes";
-import { QuestionElementType } from "../../../types/QuestionTypes";
+import {
+  QuestionElementType,
+  ResponseElementType,
+} from "../../../types/QuestionTypes";
 
-import Image from "./question-components/Image";
-import MultipleChoice from "./question-components/MultipleChoice";
-import MultiSelect from "./question-components/MultiSelect";
-import QuestionText from "./question-components/QuestionText";
-import ShortAnswer from "./question-components/ShortAnswer";
-import Text from "./question-components/Text";
+import Image from "./question-elements/Image";
+import MultipleChoice from "./question-elements/MultipleChoice";
+import MultiSelect from "./question-elements/MultiSelect";
+import QuestionText from "./question-elements/QuestionText";
+import ShortAnswer from "./question-elements/ShortAnswer";
+import Text from "./question-elements/Text";
 
-interface DisplayQuestionProps {
-  questionComponents: QuestionComponentResponse[];
+interface QuestionProps {
+  elements: QuestionComponentResponse[];
 }
 
-const Question = ({
-  questionComponents,
-}: DisplayQuestionProps): React.ReactElement => {
+const Question = ({ elements }: QuestionProps): React.ReactElement => {
+  let counter = -1;
+  const answerElementIndex = elements.map((element) => {
+    if (element.type in ResponseElementType) {
+      counter += 1;
+      return counter;
+    }
+    return 0;
+  });
+
   return (
     <>
-      {questionComponents.map((question, i) => {
+      {elements.map((element, i) => {
         return (
           <Box key={i} my={10}>
             {(() => {
-              switch (question.type) {
+              switch (element.type) {
                 case QuestionElementType.QUESTION_TEXT:
                   return (
                     <QuestionText
                       questionText={
-                        (question.metadata as QuestionTextMetadata).questionText
+                        (element.metadata as QuestionTextMetadata).questionText
```

Comment:
Also sad that we have to cast but unavoidable without a huge refactor probably 😬